packages/datadog-plugin-kafkajs/test/index.spec.js

```diff
@@ -0,0 +1,139 @@
+            try {
+              const expectedSpanPromise = expectSpanWithDefaults({
+                name: 'kafka.produce',
+                service: 'test-kafka',
+                meta: {
+                  'span.kind': 'producer',
+                  'component': 'kafka'
+                },
+                metrics: {
+                  'kafka.batch.size': messages.length
+                },
+                resource: testTopic,
+                error: 0,
+                type: 'queue'
+
+              })
+
+              await producer.connect()
+              await producer.send({
+                topic: testTopic,
+                messages
+              })
+              // Useful to debug trace
+              // agent.use(traces => console.log(traces[0]))
+
+              return expectedSpanPromise
+            } catch (error) {
+              // console.log(error)
+            }
+          })
+          it('should propagate context', async () => {
+            const producer = kafka.producer()
+            const firstSpan = tracer.scope().active()
+            await producer.connect()
+            await producer.send({
+              topic: testTopic,
+              messages: [{ key: 'key1', value: 'test' }]
+            })
+
+            return expect(tracer.scope().active()).to.equal(firstSpan)
+          })
+        })
+        describe('consumer', () => {
+          it('should be instrumented', async () => {
+            // We are changing the groupId in every test so the consumed offset is always reset
+            const consumer = kafka.consumer({ groupId: `test-group-${version}-instrument` })
+            try {
+              const expectedSpanPromise = expectSpanWithDefaults({
+                name: 'kafka.consume',
+                service: 'test-kafka',
+                meta: {
+                  'span.kind': 'consumer',
+                  'component': 'kafka'
+                },
+                resource: testTopic,
+                error: 0,
+                type: 'queue'
+              })
+
+              await consumer.connect()
+              await consumer.subscribe({ topic: testTopic, fromBeginning: true })
+              await consumer.run({
+                eachMessage: async ({ topic, partition, message }) => {
+                  expect(topic).to.equal(testTopic)
+                }
+              })
+
+              return expectedSpanPromise
+            } catch (error) {
+              // console.log(error)
+            }
+          })
+          it('should propagate context', async () => {
+            const consumer = kafka.consumer({ groupId: `test-group-${version}-propagate` })
+            const firstSpan = tracer.scope().active()
+            await consumer.connect()
+            await consumer.subscribe({ topic: testTopic, fromBeginning: true })
+            await consumer.run({
+              eachMessage: async ({ topic, partition, message }) => {
+                expect(topic).to.equal(testTopic)
```

Comment:
This will run outside the context of the test.

Reply:
There should also be an additional expectation that this is running in the context of the message span.